localboot/grub.go

```diff
@@ -74,56 +132,113 @@ func ParseGrubCfg(grubcfg string, basedir string, grubVersion int) []bootconfig.
 			if sline[0] == "linux" || sline[0] == "linux16" || sline[0] == "linuxefi" {
 				kernel := sline[1]
 				cmdline := strings.Join(sline[2:], " ")
-				if grubVersion == 2 {
-					// if grub2, unquote the string, as directives could be quoted
-					// https://www.gnu.org/software/grub/manual/grub/grub.html#Quoting
-					// TODO unquote everything, not just \$
-					cmdline = strings.Replace(cmdline, `\$`, "$", -1)
-				}
-				cfg.Kernel = path.Join(basedir, kernel)
+				cmdline = unquote(ver, cmdline)
+				cfg.Kernel = path.Join(kernelBasedir, kernel)
 				cfg.KernelArgs = cmdline
 			} else if sline[0] == "initrd" || sline[0] == "initrd16" || sline[0] == "initrdefi" {
 				initrd := sline[1]
-				cfg.Initramfs = path.Join(basedir, initrd)
+				cfg.Initramfs = path.Join(kernelBasedir, initrd)
+			} else if sline[0] == "multiboot" || sline[0] == "multiboot2" {
+				multiboot := sline[1]
+				cmdline := strings.Join(sline[2:], " ")
+				cmdline = unquote(ver, cmdline)
+				cfg.Multiboot = path.Join(kernelBasedir, multiboot)
+				cfg.MultibootArgs = cmdline
+			} else if sline[0] == "module" || sline[0] == "module2" {
+				module := sline[1]
+				cmdline := strings.Join(sline[2:], " ")
+				cmdline = unquote(ver, cmdline)
+				module = path.Join(kernelBasedir, module)
+				if cmdline != "" {
+					module = module + " " + cmdline
+				}
+				cfg.Modules = append(cfg.Modules, module)
 			}
 		}
 	}
+
 	// append last kernel config if it wasn't already
 	if inMenuEntry && cfg.IsValid() {
 		bootconfigs = append(bootconfigs, *cfg)
 	}
 	return bootconfigs
 }
 
+func isValidFsUUID(uuid string) bool {
```

Comment:
This will probably return true for lots of things we don't regard as valid UUIDs, for example it will fail to parse:
`search --set=root --label abcd`

Then again, booting with labels is not support right now anyway, and we are going to revamp the grub parser, this is fine for now.